projects/fal/src/fal/toolkit/file/file.py

```diff
@@ -181,35 +167,31 @@ def save(self, path: str | Path, overwrite: bool = False) -> Path:
 
 
 class CompressedFile(File):
-    _extract_dir: Optional[TemporaryDirectory] = PrivateAttr(default=None)
-
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self._extract_dir = None
+    extract_dir: Optional[str] = Field(default=None, exclude=True, repr=False)
 
     def __iter__(self):
-        if not self._extract_dir:
+        if not self.extract_dir:
             self._extract_files()
 
-        files = Path(self._extract_dir.name).iterdir()  # type: ignore
+        files = Path(self.extract_dir).iterdir()  # type: ignore
         return iter(files)
 
     def _extract_files(self):
-        self._extract_dir = TemporaryDirectory()
```

Comment:
Btw we were not using `delete=False` here, so this could've been deleted by the time we use glob or iter 🤷